os_net_config/utils.py

```diff
@@ -606,6 +606,88 @@ def remove_sriov_entries_for_pf(pfname):
         common.write_yaml_config(common.SRIOV_CONFIG_FILE, new_map)
 
 
+def get_sriov_dev_from_pci_address(pci_address):
+    sriov_map = common.get_sriov_map()
+    for item in sriov_map:
+        if item['pci_address'] == pci_address:
+            return item
+    return None
+
+
+def get_sriov_dev_from_name(dev_name):
+    sriov_map = common.get_sriov_map()
+    for item in sriov_map:
+        if item['name'] == dev_name:
+            return item
+    return None
+
+
+def remove_entries_for_sriov_dev(identifier):
+    """Remove SR-IOV mapping entries for a PF and/or VF.
+
+    - If a VF is specified/resolved, remove only that VF mapping entry.
+    - If only a PF is specified/resolved, remove that PF entry and all of its
```

Comment:
If a VF identifier is provided, remove the PF and all its VF entries from sriov map.